crates/usvg/src/text/mod.rs

```diff
@@ -14,16 +17,38 @@ pub mod layout;
 /// SVG specifiation. While doing so, we also calculate the text bbox (which is not based on the
 /// outlines of a glyph, but instead the glyph metrics as well as decoration spans).
 /// 2. We convert all of the positioned glyphs into outlines.
-pub(crate) fn convert(text: &mut Text, fontdb: &fontdb::Database) -> Option<()> {
-    let (text_fragments, bbox) = layout::layout_text(text, fontdb)?;
+pub(crate) fn convert(text: &mut Text, font_provider: &dyn FontProvider) -> Option<()> {
+    let (text_fragments, bbox) = layout::layout_text(text, font_provider)?;
     text.layouted = text_fragments;
     text.bounding_box = bbox.to_rect();
     text.abs_bounding_box = bbox.transform(text.abs_transform)?.to_rect();
 
-    let (group, stroke_bbox) = flatten::flatten(text, fontdb)?;
+    let (group, stroke_bbox) = flatten::flatten(text, font_provider)?;
     text.flattened = Box::new(group);
     text.stroke_bounding_box = stroke_bbox.to_rect();
     text.abs_stroke_bounding_box = stroke_bbox.transform(text.abs_transform)?.to_rect();
 
     Some(())
 }
+
+pub trait FontProvider {
+    fn resolve_font(&self, font: &Font) -> Option<ResolvedFont>;
```

Comment:
Why we return `ResolvedFont` and not just `ID`?

Reply:
And I think we should call it `find_font`. Similar to the method below.

Reply:
Good idea, will change.